sdk/textanalytics/ai-text-analytics/review/ai-text-analytics.api.md

```diff
@@ -0,0 +1,792 @@
+## API Report File for "@azure/ai-text-analytics"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AzureKeyCredential } from '@azure/core-auth';
+import { CommonClientOptions } from '@azure/core-client';
+import { KeyCredential } from '@azure/core-auth';
+import { OperationOptions } from '@azure/core-client';
+import { TokenCredential } from '@azure/core-auth';
+
+// @public
+export interface ActionCommon {
+    disableServiceLogs?: boolean;
+}
+
+// @public
+export type ActionPrebuilt = ActionCommon & {
+    modelVersion?: string;
+};
+
+// @public
+export const AnalyzeActionName: readonly ["EntityLinking", "EntityRecognition", "KeyPhraseExtraction", "PiiEntityRecognition", "LanguageDetection", "SentimentAnalysis"];
+
+// @public
+export type AnalyzeActionName = typeof AnalyzeActionName[number];
+
+// @public
+export type AnalyzeActionParameters<ActionName extends AnalyzeActionName> = ActionName extends "EntityLinking" ? EntityLinkingAction : ActionName extends "EntityRecognition" ? EntityRecognitionAction : ActionName extends "KeyPhraseExtraction" ? KeyPhraseExtractionAction : ActionName extends "PiiEntityRecognition" ? PiiEntityRecognitionAction : ActionName extends "SentimentAnalysis" ? SentimentAnalysisAction : ActionName extends "LanguageDetection" ? LanguageDetectionAction : never;
+
+// @public
+export type AnalyzeResult<ActionName extends AnalyzeActionName> = ActionName extends "EntityLinking" ? EntityLinkingResult[] : ActionName extends "EntityRecognition" ? EntityRecognitionResult[] : ActionName extends "KeyPhraseExtraction" ? KeyPhraseExtractionResult[] : ActionName extends "PiiEntityRecognition" ? PiiEntityRecognitionResult[] : ActionName extends "SentimentAnalysis" ? SentimentAnalysisResult[] : ActionName extends "LanguageDetection" ? LanguageDetectionResult[] : never;
+
+// @public
+export interface AssessmentSentiment extends SentenceAssessment {
+}
+
+export { AzureKeyCredential }
+
+// @public
+export const CognitiveLanguageApiVersion: {
+    readonly "2022-02-01-preview": "2022-02-01-preview";
+    readonly "2022-03-01-preview": "2022-03-01-preview";
```

Comment:
Is JS going to support multiple versions?

Reply:
Yes, that is the plan. Anything to be worried about?

Reply:
more like a surprise as JS has always said no todoing multiple versions. What changed?

Reply:
We have supported multiple service versions in some SDKs, like keyvault: https://cs.github.com/Azure/azure-sdk-for-js/blob/b835e0bc706dbc930657c32422e5c265a125f002/sdk/keyvault/keyvault-certificates/src/certificatesModels.ts#L24

I think it's more that we are comfortable dropping support for older service versions when there are dramatic differences in models, since we have the ability to ship new majors side-by-side.

Reply:
@witemple-msft good point, [the swagger currently does not model the list of API versions](https://github.com/Azure/azure-rest-api-specs/blob/a262813e7053f60241bc043e9899ee349b7523ba/specification/cognitiveservices/data-plane/Language/preview/2022-04-01-preview/common.json#L216), so perhaps I shouldn't add support for multi-versions support.

@assafi and @abhahn, I see that the [API Version parameter](https://github.com/Azure/azure-rest-api-specs/blob/a262813e7053f60241bc043e9899ee349b7523ba/specification/cognitiveservices/data-plane/Language/preview/2022-04-01-preview/common.json#L216) is defined as a string in common.json but I wonder if that could be moved to the TA swagger with a list of versions supported. This way the swagger guarantees that there are no breaking changes between those versions and the client libraries can follow suit and support those versions as well.

Reply:
@deyaaeldeen , since each different Swagger spec represents a single API version it seems a little awkward to specify all of the supported versions in every Swagger definition.  

We also rely on the API version parameter for query parameter versioning in the APIM configuration, which is directly based on the Swagger.  On top of this, sometimes we have to manually merge Swagger files for multiple services to create a new APIM configuration when we have multiple services releasing in the same version, so this might cause some weird behavior for us, or just be another special case for making manual changes that everyone on the team needs to be aware of which could lead to confusion.

Personally I think it makes more sense for the Language services to unify on API versions which would prevent this from being an issue, but that's just my 2 cents.  :)

Reply:
> since each different Swagger spec represents a single API version it seems a little awkward to specify all of the supported versions in every Swagger definition.

I am not sure if I follow, why would that be awkward? It is just an [enum](https://github.com/Azure/azure-rest-api-specs/blob/a74d241b28124f9ac55fef2d3a124a2be6a1eb4d/specification/cognitiveservices/data-plane/FormRecognizer/preview/2022-01-30-preview/FormRecognizer.json#L811).

> We also rely on the API version parameter for query parameter versioning in the APIM configuration, which is directly based on the Swagger

Could you help me understand what does this imply? I am not sure if I follow.

> On top of this, sometimes we have to manually merge Swagger files for multiple services to create a new APIM configuration when we have multiple services releasing in the same version, so this might cause some weird behavior for us, or just be another special case for making manual changes that everyone on the team needs to be aware of which could lead to confusion.

Having an enum for supported versions in the swagger is nothing but a contract that promises that the service team is making sure there are no breaking changes between versions, and I wonder why such an enum might affect any behavior.

Reply:
What I mean by "awkward" is that I don't think it makes sense to have an enum containing every API version attached to a single spec, which overall only represents a single API version.  Also, how would this change be governed when a new API version is released -- would we only add the new value to the enum in the newly released version, or would we have to retroactively apply this new version to all previous Swagger specs as well?

For the second part about APIM, the problem arises because we are shipping all of the services separately in API specs, but together in APIM which requires us to manually merge all of the configs.  If every service has a different subset of API versions in their enum (which they would, as of today) then we either have to find a way to merge this value or just massage it out of the spec for creating the APIM configuration (which is based directly on the Swagger).  We've just had a lot of issues with this on the service side for the past few months, and I am advocating for us to avoid more special cases if possible.